framework/db/QueryBuilder.php

```diff
@@ -278,6 +278,9 @@ public function batchInsert($table, $columns, $rows)
                 }
                 if (is_string($value)) {
                     $value = $schema->quoteValue($value);
+                } elseif (is_float($value)) {
+                    // ensure type cast always has . as decimal separator in all locales
+                    $value = str_replace(',', '.', (string) $value);
```

Comment:
Are you sure there are no separators beside `.` and `,`? I'd try to get it from `LC_NUMERIC` instead of hardcoding...